File: backend/app/domain/entities/datos_intervalo_participante.py
from dataclasses import dataclass
from typing import Optional
from datetime import datetime

@dataclass
class DatosIntervaloParticipanteEntity:
    idDatosIntervaloParticipante: Optional[int] = None
    timestamp: Optional[datetime] = None
    consumoReal_kWh: Optional[float] = None
    autoconsumo_kWh: Optional[float] = None
    energiaRecibidaReparto_kWh: Optional[float] = None
    energiaAlmacenamiento_kWh: Optional[float] = None
    energiaDiferencia_kWh: Optional[float] = None
    excedenteVertidoCompensado_kWh: Optional[float] = None
    precioImportacionIntervalo: Optional[float] = None
    precioExportacionIntervalo: Optional[float] = None
    idResultadoParticipante: Optional[int] = None
    

File: frontend/lib/models/datos_intervalo_participante.dart
import 'package:json_annotation/json_annotation.dart';

part 'datos_intervalo_participante.g.dart';

@JsonSerializable()
class DatosIntervaloParticipante {
  final int? idDatosIntervaloParticipante;
  final DateTime? timestamp;
  final double? consumoReal_kWh;
  final double? autoconsumo_kWh;
  final double? energiaRecibidaReparto_kWh;
  final double? energiaAlmacenamiento_kWh;
  final double? energiaDiferencia_kWh;
  final double? excedenteVertidoCompensado_kWh;
  final double? precioImportacionIntervalo;
  final double? precioExportacionIntervalo;
  final int? idResultadoParticipante;

  DatosIntervaloParticipante({
    this.idDatosIntervaloParticipante,
    this.timestamp,
    this.consumoReal_kWh,
    this.autoconsumo_kWh,
    this.energiaRecibidaReparto_kWh,
    this.energiaAlmacenamiento_kWh,
    this.energiaDiferencia_kWh,
    this.excedenteVertidoCompensado_kWh,
    this.precioImportacionIntervalo,
    this.precioExportacionIntervalo,
    this.idResultadoParticipante,
  });

  factory DatosIntervaloParticipante.fromJson(Map<String, dynamic> json) => _$DatosIntervaloParticipanteFromJson(json);
  Map<String, dynamic> toJson() => _$DatosIntervaloParticipanteToJson(this);
}
